agent/grpc-internal/resolver/resolver.go

```diff
@@ -210,8 +268,28 @@ func (s *ServerResolverBuilder) UpdateLeaderAddr(datacenter, addr string) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	s.leaderResolver.globalAddr = DCPrefix(datacenter, addr)
-	s.leaderResolver.updateClientConn()
+	lanAddr := DCPrefix(datacenter, addr)
+
+	s.leaderResolver.globalAddr = lanAddr
```

Comment:
Is it safe to update `globalAddr` but not call `updateClientConn`? What happens if this is in the if-block below?

Reply:
Backed out these changes as the router update changes in `server_serf.go` largely remove the need.